### ISSUE TITLE
search only returns 20 entries, even if there are more

### ISSUE THREAD

**Issue by evgeni** [SatelliteQE/nailgun#429]
Ohai,

currently, when I call `SomeEntity.search()`, I will get the first 20 entities that match the search, but not all. This is due to how Foreman does paging on the API results. While it is OK for an API to return paged results, I think `nailgun` as a client should fetch all results (by fetching all the pages and merging the results).

**Comment by Fobhep** [comment]
+1

**Comment by abalakh** [comment]
Took a look at api docs, found such parameter:
```
full_result optional | Whether or not to show all results
```
Just curious, why not to use it like `.search(query={'full_result': True})`?

**Comment by evgeni** [comment]
Does not work for Provisioning Templates.

**Comment by Fobhep** [comment]
agreed. That does not show the full result list for Provisioning Templates or PartitionTables

**Comment by evgeni** [comment]
`full_result=true` only works on Katello API, not the Foreman API.
